src/Illuminate/Bus/Queueable.php

```diff
@@ -188,9 +195,21 @@ public function dispatchNextJobInChain()
                 $next->onConnection($next->connection ?: $this->chainConnection);
                 $next->onQueue($next->queue ?: $this->chainQueue);
 
+                $next->chainCatchCallbacks = $this->chainCatchCallbacks;
                 $next->chainConnection = $this->chainConnection;
                 $next->chainQueue = $this->chainQueue;
             }));
         }
     }
+
+    /**
+     * Dispatch the next job on the chain.
+     *
+     * @param  \Throwable  $e
+     * @return void
+     */
+    public function invokeChainCatchCallbacks($e)
+    {
+        collect($this->chainCatchCallbacks)->each->__invoke($e);
```

Comment:
I don't think userland code should ever call dynamic methods like that? Can we not just literally invoke it?

```
(collect($this->chainCatchCallbacks)->each)();
```

Reply:
@GrahamCampbell hmmm what's the reason?

Reply:
Magic methods are not meant to be called directly. It's hacky to do that. Just like it's hacky to use reflection to make a private method callable and then call it. Those methods are there to be used by PHP native syntax. `__invoke` is meant to be called only be actually calling the class and `__get` is meant to be called only by doing a property access, etc. Obviously, there are exceptions, but I don't think this is one of those cases.